lighthouse-core/test/audits/byte-efficiency/offscreen-images-test.js

```diff
@@ -121,189 +106,208 @@ describe('OffscreenImages audit', () => {
+    const wastedSize = 100 * 1024;
+    const recordA = {
+      url: 'https://example.com/a',
+      resourceSize: wastedSize,
+      requestId: 'a',
+      startTime: 1,
+      endTime: 1.25,
+      priority: 'High',
+      timing: {receiveHeadersEnd: 1.25},
+    };
+    const recordB = {
+      url: 'https://example.com/b',
+      resourceSize: wastedSize,
+      requestId: 'b',
+      startTime: 2.25,
+      endTime: 2.5,
+      priority: 'High',
+      timing: {receiveHeadersEnd: 2.5},
+    };
+    const devtoolsLog = networkRecordsToDevtoolsLog([recordA, recordB]);
+
+    const topLevelTasks = [
+      {ts: 1975, duration: 50},
+    ];
+    const artifacts = Object.assign(Runner.instantiateComputedArtifacts(), {
       ViewportDimensions: DEFAULT_DIMENSIONS,
       ImageUsage: [
         generateImage(generateSize(0, 0), [0, 0], recordA, recordA.url),
         generateImage(generateSize(200, 200), [3000, 0], recordB, recordB.url),
       ],
-      traces: {},
-      devtoolsLogs: {},
-      requestInteractive: async () => ({pessimisticEstimate: {nodeTimings: timings}}),
-    }, [], context, [], context).then(auditResult => {
+      traces: {defaultPass: createTestTrace({topLevelTasks})},
+      devtoolsLogs: {defaultPass: devtoolsLog},
+    });
+
+    return UnusedImages.audit_(artifacts, [], context).then(auditResult => {
       assert.equal(auditResult.items.length, 1);
-      assert.equal(auditResult.items[0].url, 'a');
+      assert.equal(auditResult.items[0].url, recordA.url);
+      assert.equal(auditResult.items[0].wastedBytes, wastedSize);
     });
   });
 
   it('finds images loaded before last long task (Lantern)', () => {
     context = {settings: {throttlingMethod: 'simulate'}};
-    const recordA = {url: 'a', resourceSize: 100 * 1024, requestId: 'a'};
-    const recordB = {url: 'b', resourceSize: 100 * 1024, requestId: 'b'};
-
-    const networkA = new NetworkNode(recordA);
-    const networkB = new NetworkNode(recordB);
-    const cpu = new CPUNode({}, []);
-    const timings = new Map([
-      [networkA, {startTime: 1000}],
-      [networkB, {startTime: 1500}],
-      [cpu, {startTime: 1975, endTime: 2025, duration: 50}],
-    ]);
-
-    return UnusedImages.audit_({
+    const wastedSize = 100 * 1024;
+    const recordA = {
+      url: 'https://example.com/a',
+      resourceSize: wastedSize,
+      requestId: 'a',
+      startTime: 1,
+      endTime: 1.25,
+      priority: 'High',
+      timing: {receiveHeadersEnd: 1.25},
+    };
+    const recordB = {
+      url: 'https://example.com/b',
+      resourceSize: wastedSize,
+      requestId: 'b',
+      startTime: 1.25,
+      endTime: 1.5,
+      priority: 'High',
+      timing: {receiveHeadersEnd: 1.5},
+    };
+    const devtoolsLog = networkRecordsToDevtoolsLog([recordA, recordB]);
+
+    // Enough tasks to spread out graph.
+    const topLevelTasks = [
+      {ts: 1000, duration: 10},
+      {ts: 1050, duration: 10},
+      {ts: 1975, duration: 50},
```

Comment:
@patrickhulce I don't entirely understand why I couldn't get the pessimistic graph to download these images in parallel without these extra 10ms tasks. Instead the second image and the CPU node always went in parallel (even if the two images had identical start and end times) and so it was always failing seeing both of these images before the last long task started.

Inserting these extra top level tasks fixed that and doesn't feel too bad (most traces have more than one top level task :), but I do worry about the fragility of my solution

Reply:
I can't tell for 100% just from glancing, but it's probably because they were inferred to have come from the same connection, if different connectionIds are added and/or the timings show they happened in parallel then it might fix itself?

Reply:
> different connectionIds are added and/or the timings show they happened in parallel 

ah, didn't work. There's probably some subtlety with leaving out real connection timings or something, too. I'll just leave it like this for now and maybe we'll figure it out when a lantern change breaks this test :)